account_invoice_merge/invoice.py

```diff
@@ -166,7 +170,7 @@ def make_key(br, fields):
                 (0, 0, value) for value in
                 invoice_data['invoice_line'].itervalues()]
             # create the new invoice
-            newinvoice = self.create(invoice_data)
+            newinvoice = self.with_context(context).create(invoice_data)
```

Comment:
It can be used in both ways: passing a dictionary to replace the current one, or only passing keyword arguments to extend current one.